CollectNewspaperKarel.java

```diff
@@ -1,17 +1,43 @@
-/*
- * File: CollectNewspaperKarel.java
- * --------------------------------
- * At present, the CollectNewspaperKarel subclass does nothing.
- * Your job in the assignment is to add the necessary code to
- * instruct Karel to walk to the door of its house, pick up the
- * newspaper (represented by a beeper, of course), and then return
- * to its initial position in the upper left corner of the house.
- */
-
-import stanford.karel.*;
-
-public class CollectNewspaperKarel extends SuperKarel {
-
-	// You fill in this part
-
-}
+/*
+ * File: CollectNewspaperKarel.java
+ * --------------------------------
+ * At present, the CollectNewspaperKarel subclass does nothing.
+ * Your job in the assignment is to add the necessary code to
+ * instruct Karel to walk to the door of its house, pick up the
+ * newspaper (represented by a beeper, of course), and then return
+ * to its initial position in the upper left corner of the house.
+ */
+
+import stanford.karel.*;
+
+public class CollectNewspaperKarel extends SuperKarel {
+
+	public void run() {

```

Comment:
Your code works but you should have broken it down to make it easier to understand. For example you could have had two methods that you could call from the run method follows:

```
public void run() {
    pickUpNewsPaper();
    returnToStartPosition();
}
```